w3af/plugins/grep/cdn_providers.py

```diff
@@ -0,0 +1,255 @@
+        ["server", "cloudflare", "Cloudflare"],
+        ["server", "yunjiasu", "Yunjiasu"],
+        ["server", "ECS", "Edgecast"],
+        ["server", "ECAcc", "Edgecast"],
+        ["server", "ECD", "Edgecast"],
+        ["server", "NetDNA", "NetDNA"],
+        ["server", "Airee", "Airee"],
+        ["X-CDN-Geo", "", "OVH CDN"],
+        ["X-CDN-Pop", "", "OVH CDN"],
+        ["X-Px", "", "CDNetworks"],
+        ["X-Instart-Request-ID", "instart", "Instart Logic"],
+        ["Via", "CloudFront", "Amazon CloudFront"],
+        ["X-Edge-IP", "", "CDN"],
+        ["X-Edge-Location", "", "CDN"],
+        ["X-HW", "", "Highwinds"],
+        ["X-Powered-By", "NYI FTW", "NYI FTW"],
+        ["X-Delivered-By", "NYI FTW", "NYI FTW"],
+        ["server", "ReSRC", "ReSRC.it"],
+        ["X-Cdn", "Zenedge", "Zenedge"],
+        ["server", "leasewebcdn", "LeaseWeb CDN"],
+        ["Via", "Rev-Cache", "Rev Software"],
+        ["X-Rev-Cache", "", "Rev Software"],
+        ["Server", "Caspowa", "Caspowa"],
+        ["Server", "SurgeCDN", "Surge"],
+        ["server", "sffe", "Google"],
+        ["server", "gws", "Google"],
+        ["server", "GSE", "Google"],
+        ["server", "Golfe2", "Google"],
+        ["Via", "google", "Google"],
+        ["server", "tsa_b", "Twitter"],
+        ["X-Cache", "cache.51cdn.com", "ChinaNetCenter"],
+        ["X-CDN", "Incapsula", "Incapsula"],
+        ["X-Iinfo", "", "Incapsula"],
+        ["X-Ar-Debug", "", "Aryaka"],
+        ["server", "gocache", "GoCache"],
+        ["server", "hiberniacdn", "HiberniaCDN"],
+        ["server", "UnicornCDN", "UnicornCDN"],
+        ["server", "Optimal CDN", "Optimal CDN"],
+        ["server", "Sucuri/Cloudproxy", "Sucuri Firewall"],
+        ["x-sucuri-id", "", "Sucuri Firewall"],
+        ["server", "Netlify", "Netlify"],
+        ["section-io-id", "", "section.io"],
+        ["server", "Testa/", "Naver"],
+        ["server", "BunnyCDN", "BunnyCDN"],
+        ["server", "MNCDN", "Medianova"],
+        ["server", "Roast.io", "Roast.io"],
+        ["x-rocket-node", "", "Rocket CDN"],
+    )
+
+    # CDN domains stored in format ['required part of a domain', 'provider's name']
+    cdn_domains = [
+        [".akamai.net", "Akamai"],
+        [".akamaized.net", "Akamai"],
+        [".akamaiedge.net", "Akamai"],
+        [".akamaihd.net", "Akamai"],
+        [".edgesuite.net", "Akamai"],
+        [".edgekey.net", "Akamai"],
+        [".srip.net", "Akamai"],
+        [".akamaitechnologies.com", "Akamai"],
+        [".akamaitechnologies.fr", "Akamai"],
+        [".tl88.net", "Akamai China CDN"],
+        [".llnwd.net", "Limelight"],
+        ["edgecastcdn.net", "Edgecast"],
+        [".systemcdn.net", "Edgecast"],
+        [".transactcdn.net", "Edgecast"],
+        [".v1cdn.net", "Edgecast"],
+        [".v2cdn.net", "Edgecast"],
+        [".v3cdn.net", "Edgecast"],
+        [".v4cdn.net", "Edgecast"],
+        [".v5cdn.net", "Edgecast"],
+        ["hwcdn.net", "Highwinds"],
```

Comment:
Missing leading dot here? Or all the others don't need the leading dot?

Reply:
I guess Highwinds is discovered only when we can't see any subdomain in url, while for example Netlify's url may look like `cdn.netlify.com`, so we get here `.netlify.com` to make subdomain is used.